### PR TITLE
Support SplashAttention in PyTorch/XLA

### PR DESCRIPTION
**Pull request by wonjoo-wj**
Support SplashAttention in PyTorch/XLA

References of SplashAttention in JAX:
- SplashAttention implementation: [jax/experimental/pallas/ops/tpu/splash_attention](https://github.com/google/jax/tree/main/jax/experimental/pallas/ops/tpu/splash_attention)
- SplashAttention tests
  - Kernel test: [tests/pallas/tpu_splash_attention_kernel_test.py](https://github.com/google/jax/blob/main/tests/pallas/tpu_splash_attention_kernel_test.py)
  - Mask test: [tests/pallas/tpu_splash_attention_mask_test.py](https://github.com/google/jax/blob/main/tests/pallas/tpu_splash_attention_mask_test.py)

### CODE REVIEW

**Comment by wonjoolee95** [comment]
As soon as the CI is green, this should to be ready to reviewed.

**Comment by miladm** [comment]
@alanwaketan @wonjoolee95 how close are we to landing this PR?
